From baeb5aa827d956bd06492775dc5bd9f89d394149 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 13 Jul 2015 23:38:41 -0400
Subject: [PATCH] add configure flag to disable tests

The tests require the check package to be installed.  There is a configure
test for this package (which is good), but it's unconditional (which is
bad) as it means you can't even build & install kbd w/out the check
package being installed.

URL: https://bugs.gentoo.org/485116
Reported-by: Arfrever Frehtes Taifersar Arahesis <arfrever.fta@gmail.com>
---
 Makefile.am  |  5 ++++-
 configure.ac | 17 +++++++++++++++--
 2 files changed, 19 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 89c7e83..f855110 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,7 +8,10 @@ EXTRA_DIST = \
 	CREDITS \
 	contrib docs rc
 
-SUBDIRS = src data po tests docs
+SUBDIRS = src data po docs
+if BUILD_TESTS
+SUBDIRS += tests
+endif
 
 kbd-$(VERSION).tar.xz:
 	make distcheck
diff --git a/configure.ac b/configure.ac
index d0b462b..f659758 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,8 +10,21 @@ AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([1.9 -Wall color-tests dist-xz])
 AC_CONFIG_SRCDIR([src/loadkeys.c])
 AC_CONFIG_HEADERS(config.h)
-
-PKG_CHECK_MODULES([CHECK], [check >= 0.9.4])
+PKG_PROG_PKG_CONFIG
+
+AC_ARG_ENABLE(tests,
+	[AS_HELP_STRING([--disable-tests], [do not build tests])],
+	[build_tests=$enableval], [build_tests=auto])
+if test "$build_tests" != "no"; then
+	PKG_CHECK_MODULES([CHECK], [check >= 0.9.4],
+		[build_tests="yes"], [
+			if test "$build_tests" = "yes"; then
+				AC_MSG_ERROR([tests requested, but check package is missing])
+			fi
+			build_tests="no"
+		])
+fi
+AM_CONDITIONAL(BUILD_TESTS, test "$build_tests" != "no")
 
 m4_ifndef([AM_SILENT_RULES], [m4_define([AM_SILENT_RULES],[])])
 AM_SILENT_RULES([yes])
-- 
2.6.4

